cognite/client/_api/datapoints.py

```diff
@@ -769,18 +792,29 @@ def retrieve(
             external_id=external_id,
             aggregates=aggregates,
             granularity=granularity,
+            target_unit=target_unit,
+            target_unit_system=target_unit_system,
             limit=limit,
             include_outside_points=include_outside_points,
             ignore_unknown_ids=ignore_unknown_ids,
         )
         fetcher = select_dps_fetch_strategy(self, user_query=query)
-        dps_lst = fetcher.fetch_all_datapoints()
+        api_subversion = self._get_api_subversion(target_unit, target_unit_system)
+
+        dps_lst = fetcher.fetch_all_datapoints(api_subversion)
         if not query.is_single_identifier:
             return dps_lst
         elif not dps_lst and ignore_unknown_ids:
             return None
         return dps_lst[0]
 
+    def _get_api_subversion(self, target_unit: str | None, target_unit_system: str | None) -> str | None:
+        api_subversion = None
+        if target_unit is not None or target_unit_system is not None:
+            api_subversion = "beta"
+            self._unit_warning.warn()
+        return api_subversion
```

Comment:
See prev. comment for context. 

This would need to be more involved and check the per-time-series settings.

Reply:
Also fixed in 9b6accd.